RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitPhase2WeightsKernels.cu

```diff
@@ -0,0 +1,62 @@
+#include <cuda.h>
+
+#include "FWCore/Utilities/interface/CMSUnrollLoop.h"
+#include "DataFormats/EcalRecHit/interface/EcalUncalibratedRecHit.h"
+#include "DataFormats/EcalDigi/interface/EcalLiteDTUSample.h"
+#include "DataFormats/EcalDigi/interface/EcalConstants.h"
+
+#include "EcalUncalibRecHitPhase2WeightsKernels.h"
+
+namespace ecal {
+  namespace weights {
+
+    __global__ void Phase2WeightsKernel(uint16_t const* digis_in,
+                                        uint32_t const* dids,
+                                        ::ecal::reco::StorageScalarType* amplitude,
+                                        ::ecal::reco::StorageScalarType* amplitudeError,
+                                        uint32_t* dids_out,
+                                        int const nchannels,
+                                        double* weights,
+                                        uint32_t* flags) {
+      constexpr int nsamples = EcalDataFrame_Ph2::MAXSAMPLES;
+      int const tx = threadIdx.x + blockIdx.x * blockDim.x;
+      unsigned int nchannels_per_block = blockDim.x;
+
+      if (tx < nchannels) {
```

Comment:
I think the part that copies data from global to shared memory should run also when `tx >= nchannels` ?